worlds/sc2/PoolFilter.py

```diff
@@ -8,21 +8,25 @@
 from .Options import get_option_value, MissionOrder, \
     get_enabled_campaigns, get_disabled_campaigns, RequiredTactics, kerrigan_unit_available
 from .LogicMixin import SC2Logic
+from . import ItemNames
 
 # Items with associated upgrades
-UPGRADABLE_ITEMS = [
-    # WoL
-    "Marine", "Medic", "Firebat", "Marauder", "Reaper", "Ghost", "Spectre",
-    "Hellion", "Vulture", "Goliath", "Diamondback", "Siege Tank", "Thor", "Predator", "Widow Mine", "Cyclone",
-    "Medivac", "Wraith", "Viking", "Banshee", "Battlecruiser", "Raven", "Science Vessel", "Liberator", "Valkyrie",
-    "Bunker", "Missile Turret",
-    # HotS
-    "Zergling", "Roach", "Hydralisk", "Baneling", "Mutalisk", "Swarm Host", "Ultralisk"
-]
-
-BARRACKS_UNITS = {"Marine", "Medic", "Firebat", "Marauder", "Reaper", "Ghost", "Spectre"}
-FACTORY_UNITS = {"Hellion", "Vulture", "Goliath", "Diamondback", "Siege Tank", "Thor", "Predator", "Widow Mine", "Cyclone"}
-STARPORT_UNITS = {"Medivac", "Wraith", "Viking", "Banshee", "Battlecruiser", "Hercules", "Science Vessel", "Raven", "Liberator", "Valkyrie"}
+UPGRADABLE_ITEMS = {item.parent_item for item in get_full_item_list().values() if item.parent_item}
```

Comment:
I breakpointed in, and this set is equivalent to the old list (same length and contents). This object is only used with the `in` operator, so there's no need to keep it a list. It's more expensive, but this comprehension is only done once and is more future-proof as new upgrades are added.

I'm tempted to say this and the three sets below should probably live in another file; possibly a new one with item descriptions once those are added.